libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -271,7 +276,7 @@ proc connectOnce(p: PubSubPeer): Future[void] {.async.} =
   finally:
     await p.closeSendConn(PubSubPeerEventKind.StreamClosed)
 
-proc connectImpl(p: PubSubPeer) {.async.} =
+proc connectImpl(p: PubSubPeer) {.async: (raises: []).} =
```

Comment:
this _looks like_ it should propagate cancellederror  - "most" api should do this to support cancellation - is there a specific reason here not to?

Reply:
Good question. The prev version had an `except CatchableError` that captures the `CancelledError` and also interestingly enoug a comment that says `never cancelled` https://github.com/vacp2p/nim-libp2p/blob/1fa30f07e81e9ef415a83c5d3846e86e99a7857f/libp2p/protocols/pubsub/pubsubpeer.nim#L285-L286 

Looking at the history, you introduced this change in https://github.com/vacp2p/nim-libp2p/commit/70deac9e0d16f7d00a0d4404ed191042dbf079be#diff-6f99d21f9b0d636313f9633209e5b125903440de8a134b40821e70c9b9ae4591 as it previously raised the `CancelledError`. Do you remember why?
![image](https://github.com/user-attachments/assets/c97d9875-4243-4aad-90e2-3af2a8812c8c)